packages/website/src/docs/extending-cadl/basics.md

```diff
@@ -0,0 +1,161 @@
+
+```json
+  "main": "dist/index.js",
+  "cadlMain": "lib/main.cadl"
+```
+
+### 3. Install and initialize TypeScript
+
+Run the following commands:
+
+```bash
+> npm install -D typescript
+> npx tsc --init --strict
+```
+
+This will create `tsconfig.json`. But we need to make a couple changes to this. Open `tsconfig.json` and set the following settings:
+
+```json
+    "moduleResolution": "node",
+    "module": "ESNext",
+    "target": "es2018",
+    "rootDir": "./src",
+    "outDir": "./dist",
+```
+
+### 4. Create `lib.ts`
+
+Open `./src/lib.ts` and create your library definition that registers your library with the Cadl compiler and defines any diagnostics your library will emit. The following shows an example:
+
+```typescript
+import { createCadlLibrary } from "@cadl-lang/compiler";
+
+export const myLibrary = createCadlLibrary({
+  name: "myLibrary",
+  diagnostics: {},
+});
+
+// optional but convenient
+export const { reportDiagnostic, createDiagnostic, createStateSymbol } = myLibrary;
+```
+
+Diagnostics are used for linters and decorators which are covered in subsequent topics.
+
+### 5. Create `index.ts`
+
+Open `./src/index.ts` and import your library definition:
+
+```typescript
+import { myLibrary } from "./lib.js";
+```
+
+### 6. Build TypeScript
+
+Cadl can only import JavaScript files, so any time changes are made to TypeScript sources, they need to be compiled before they are visible to Cadl. To do so, run `npx tsc -p .` in your library's root directory. You can also run `npx tsc -p --watch` if you would like to re-run the TypeScript compiler whenever files are changed.
+
+### 7. Add your main Cadl file
+
+Open `./lib/main.cadl` and import your JS entrypoint. This ensures that when cadl imports your library, the code to define the library is run. In later topics when we add decorators, this import will ensure those get exposed as well.
+
+```cadl
+import "../dist/index.js";
+```
+
+## Adding Cadl types to your library
+
+Open `./lib/main.cadl` and add any types you want to be available when users import this library. It is also strongly recommended you put these types in a namespace that corresponds with the library name. For example, your `./lib/main.cadl` file might look like:
+
+```cadl
+import "../dist/index.js";
+
+namespace MyLibrary;
+model Person {
+  name: string;
+  age: uint8;
+}
+```
+
+## Publishing your Cadl library
+
+To publish to the public npm registry, follow [their documentation](https://docs.npmjs.com/creating-and-publishing-unscoped-public-packages).
+
+## Importing your Cadl library
```

Comment:
would it be worth talking about the fact that you can self import your library(e.g. from a samples folder inside your package)